tools/c7n_azure/c7n_azure/resources/network_security_group.py

```diff
@@ -0,0 +1,207 @@
+                    return False
+        # OnlyPorts is specified, anything NOT included in OnlyPorts should return True
+        if self.only_ports:
+            for op in self.only_ports:
+                if len(dest_port_range) > 1:
+                    if dest_port_range[0] <= op >= dest_port_range[1]:
+                        return False
+                else:
+                    if dest_port_range[0] == op:
+                        return False
+        # Ports is specified, only those included in Ports should return true
+        elif self.ports:
+            if len(dest_port_range) > 1:
+                # self.ports needs to have ALL ports in range (inclusive) to match
+                range_set = set(range(dest_port_range[0], dest_port_range[1] + 1))
+                ports_set = set(self.ports)
+                return range_set.issubset(ports_set)
+            else:
+                return dest_port_range[0] in self.ports
+        return True
+
+    def is_ranges_match(self, security_rule):
+        if 'destinationPortRange' in security_rule['properties']:
+            dest_port_ranges = \
+                [self.get_port_range(security_rule['properties']['destinationPortRange'])]
+        else:
+            dest_port_ranges = \
+                [self.get_port_range(range_str) for range_str
+                 in security_rule['properties']['destinationPortRanges']]
+        for range in dest_port_ranges:
+            if not self.is_range_match(range):
+                return False
+        return True
+
+    def get_port_range(self, range_str):
+        return [int(item) for item in range_str.split('-')]
+
+    '''
+    Determine if SecurityRule matches criteria as entered in policy
+
+    Currently supporting filters:
+        Ports
+        OnlyPorts
+        FromPort
+        ToPort
+        IpProtocol
+    '''
+
+    def is_match(self, security_rule):
+        if self.direction_key != security_rule['properties']['direction']:
+            return False
+        ranges_match = self.is_ranges_match(security_rule)
+        protocol_match = (self.ip_protocol is None) or \
+                         (self.ip_protocol == security_rule['properties']['protocol'])
+        return self.match_op([ranges_match, protocol_match])
+
+
+@NetworkSecurityGroup.filter_registry.register('ingress')
+class IngressFilter(SecurityRuleFilter):
+    direction_key = 'Inbound'
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'type': {'enum': ['ingress']},
+            'match-operator': {'type': 'string', 'enum': ['or', 'and']},
+            'Ports': {'type': 'array', 'items': {'type': 'integer'}},
+            'OnlyPorts': {'type': 'array', 'items': {'type': 'integer'}},
+            'FromPort': {'type': 'integer'},
+            'ToPort': {'type': 'integer'},
+            'IpProtocol': {'type': 'string', 'enum': ['TCP', 'UDP']}
+            # 'SelfReference': {'type': 'boolean'} TODO What does this mean in AWS?
```

Comment:
for unneeded properties we shouldn't be adding it and commenting it out, if it's unneeded we can just leave it out. Also, self reference is in reference to a security group in AWS that references itself, i.e. sg-123456 allows traffic over port 80 to sg-123456